CI (Windows): got unexpected output when executing "npm run build-setup-python"

## What happened

On the Windows runner, we got two errors (TS7016 and TS2322) during the execution of the "[npm run build](https://github.com/jurplel/install-qt-action/blob/59f08e42a726f2bd91e36457a99d34dc1aff54cb/.github/workflows/test.yml#L22-L25)" step. We can view the logs in [test (windows-2019, 5.9.0)](https://github.com/jurplel/install-qt-action/runs/3751160608?check_suite_focus=true#step:5:6) or [test (windows-2019, 5.15.1)](https://github.com/jurplel/install-qt-action/runs/3751160621?check_suite_focus=true#step:5:6):

```console
> install-qt-action@0.0.0 build-setup-python D:\a\install-qt-action\install-qt-action
> npm install --prefix node_modules/setup-python && npm run build --prefix node_modules/setup-python

+ install-qt-action@0.0.0
added 1 package from 1 contributor in 0.14s

> setup-python@2.0.1 build D:\a\install-qt-action\install-qt-action\node_modules\setup-python
> tsc

Error: src/find-python.ts(4,25): error TS7016: Could not find a declaration file for module 'semver'. 'D:/a/install-qt-action/install-qt-action/node_modules/semver/index.js' implicitly has an 'any' type.
  Try `npm install @types/semver` if it exists or add a new declaration (.d.ts) file containing `declare module 'semver';`
Error: src/install-python.ts(39,5): error TS2322: Type '{ [key: string]: string | undefined; } | { LD_LIBRARY_PATH: string; }' is not assignable to type '{ [key: string]: string; } | undefined'.
  Type '{ [key: string]: string | undefined; }' is not assignable to type '{ [key: string]: string; }'.
    Index signatures are incompatible.
      Type 'string | undefined' is not assignable to type 'string'.
        Type 'undefined' is not assignable to type 'string'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! setup-python@2.0.1 build: `tsc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the setup-python@2.0.1 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\npm\cache\_logs\...-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! install-qt-action@0.0.0 build-setup-python: `npm install --prefix node_modules/setup-python && npm run build --prefix node_modules/setup-python`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the install-qt-action@0.0.0 build-setup-python script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\npm\cache\_logs\...-debug.log

> install-qt-action@0.0.0 build D:\a\install-qt-action\install-qt-action
> tsc

```

Essentially, [two npm scripts](https://github.com/jurplel/install-qt-action/blob/59f08e42a726f2bd91e36457a99d34dc1aff54cb/package.json#L8-L9) were executed in this step:

| Code in the workflow step | npm script
| - | -
| `npm run build-setup-python` | `npm install --prefix node_modules/setup-python && npm run build --prefix node_modules/setup-python`
| `npm run build` | `tsc`

According to the output, we know that the errors occurred when executing `npm run build --prefix node_modules/setup-python` . However, the workflow was not interrupted at all, and the subsequent call to `tsc` was successful. In other words, the imported library "node_modules/setup-python/lib/find-python.js" was built successfully.

## Why was the CI workflow not interrupted?

npm exited with error code 2:

```console
...

npm ERR! install-qt-action@0.0.0 build-setup-python: `npm install --prefix node_modules/setup-python && npm run build --prefix node_modules/setup-python`
npm ERR! Exit status 2
```

We executed those commands in pwsh on Windows:

> ### [Using a specific shell](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#using-a-specific-shell)
> 
> Supported platform | shell parameter | Description | Command run internally
> -- | -- | -- | --
> Windows | pwsh | This is the default shell used on Windows. The PowerShell Core. GitHub appends the extension .ps1 to your script name. If your self-hosted Windows runner does not have PowerShell Core installed, then PowerShell Desktop is used instead. | pwsh -command ". '{0}'".

PowerShell, currently, doesn't have the ability to handle the error from native commands (executables). Therefore, if we don't check either [`$?`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables#section-1) or [`$LastExitCode`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables#lastexitcode) after executing commands with npm, the script will continue to execute to the end.

This is easy to fix. But obviously, once it is fixed, the CI workflow will fail.

See also [windows - How to stop a PowerShell script on the first error? - Stack Overflow](https://stackoverflow.com/q/9948517).

--- a/package.json
+++ b/package.json
@@ -22,7 +22,8 @@
     "setup-python": "github:actions/setup-python#v2"
   },
   "devDependencies": {
-    "@types/node": "^12.0.4",
-    "typescript": "^3.5.1"
+    "@types/node": "^12.12.31",
+    "@types/semver": "^7.1.0",
+    "typescript": "^3.8.3"
   }
 }

